commit d69ad39ad5d33b26efbbb50adb825849622f1d88
Author: Dave Martin <Dave.Martin@arm.com>
Date:   Fri Jan 4 13:09:50 2019 +0000

    arm64/sve: ptrace: Fix SVE_PT_REGS_OFFSET definition
    
    [ Upstream commit ee1b465b303591d3a04d403122bbc0d7026520fb ]
    
    SVE_PT_REGS_OFFSET is supposed to indicate the offset for skipping
    over the ptrace NT_ARM_SVE header (struct user_sve_header) to the
    start of the SVE register data proper.
    
    However, currently SVE_PT_REGS_OFFSET is defined in terms of struct
    sve_context, which is wrong: that structure describes the SVE
    header in the signal frame, not in the ptrace regset.
    
    This patch fixes the definition to use the ptrace header structure
    struct user_sve_header instead.
    
    By good fortune, the two structures are the same size anyway, so
    there is no functional or ABI change.
    
    Signed-off-by: Dave Martin <Dave.Martin@arm.com>
    Signed-off-by: Will Deacon <will.deacon@arm.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/arch/arm64/include/uapi/asm/ptrace.h b/arch/arm64/include/uapi/asm/ptrace.h
index 98c4ce55d9c3..ad64d2c92ef5 100644
--- a/arch/arm64/include/uapi/asm/ptrace.h
+++ b/arch/arm64/include/uapi/asm/ptrace.h
@@ -130,7 +130,7 @@ struct user_sve_header {
 
 /* Offset from the start of struct user_sve_header to the register data */
 #define SVE_PT_REGS_OFFSET					\
-	((sizeof(struct sve_context) + (SVE_VQ_BYTES - 1))	\
+	((sizeof(struct user_sve_header) + (SVE_VQ_BYTES - 1))	\
 		/ SVE_VQ_BYTES * SVE_VQ_BYTES)
 
 /*
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1231_d69ad39ad5d33b26efbbb50adb825849622f1d88.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1231_d69ad39ad5d33b26efbbb50adb825849622f1d88.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
